nativelib/src/main/scala/scala/scalanative/native/time.scala

```diff
@@ -0,0 +1,27 @@
+package scala.scalanative
+package native
+
+@extern
+object time {
+
+  type time_t = CLong
+
+  @struct
+  class timespec(
+      val tv_sec: time_t = 0L, /* seconds */
+      val tv_nsec: CLong = 0L /* nanoseconds */
+  )
+
+  def clock_getres(clk_id: Clock.clockid_t, res: Ptr[timespec]): CInt  = extern
+  def clock_gettime(clk_id: Clock.clockid_t, res: Ptr[timespec]): CInt = extern
+  def clock_settime(clk_id: Clock.clockid_t, res: Ptr[timespec]): CInt = extern
```

Comment:
I'll prioritise #58. Lets just have posix implementation in master for now. We probably don't want to expose those as public part of SN yet, so lets put bindings to that in `scala.scalanative.runtime` under `@extern object posix`. Just make sure that you use only cross-platform posix APIs and don't use linux-only things.